src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy

```diff
@@ -188,4 +193,37 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
         return targetFile.absolutePath
     }
 
+    private void createCheckCodeQualityErrorsTask(Project project) {
+        project.tasks.create(name: CHECK_CODE_QUALITY_ERRORS_TASK) {
+            description("Fail build if any Checkstyle, Pmd or Findbugs error found.")
+            doLast {
+                List reportErrors = []
+                project.sourceSets.each { sourceSet ->
+                    String sourceName = sourceSet.name
+                    List checkstyleReportErrors = CodeQualityUtil.getCheckstyleReportErrors(project, sourceName)
+                    if (!checkstyleReportErrors.isEmpty()) {
```

Comment:
by default `getCheckstyleReportErrors` retruns [], so we might have empty arrays  where we have no errors